File: src/types/mcp.ts
/** MCP server record managed by the local catalogue. */
export interface McpServer {
  id: string;
  name: string;
  description: string;
  command: string;
  args: string;
  env: string;
  isEnabled: boolean;
  isBuiltin: boolean;
  cliBindings: string[];
  createdAt: string;
  updatedAt: string;
}

/** Payload for creating an MCP server entry. */
export interface CreateMcpServerPayload {
  id?: string;
  name: string;
  description?: string;
  command: string;
  args?: string;
  env?: string;
  isEnabled?: boolean;
  cliBindings?: string[];
}

/** Payload for updating an MCP server entry. */
export interface UpdateMcpServerPayload {
  id: string;
  name: string;
  description?: string;
  command: string;
  args?: string;
  env?: string;
  isEnabled?: boolean;
  cliBindings?: string[];
}

File: src-tauri/src/models/mcp.rs
use serde::{Deserialize, Serialize};

/// Frontend-facing MCP server configuration.
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct McpServer {
    pub id: String,
    pub name: String,
    pub description: String,
    pub command: String,
    pub args: String,
    pub env: String,
    pub is_enabled: bool,
    pub is_builtin: bool,
    pub cli_bindings: Vec<String>,
    pub created_at: String,
    pub updated_at: String,
}
